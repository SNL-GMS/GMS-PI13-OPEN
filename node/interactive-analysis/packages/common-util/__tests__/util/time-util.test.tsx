import {
  MILLIS_IN_DAY,
  MILLIS_IN_MINUTE,
  MILLIS_IN_WEEK,
  MILLISECONDS_IN_SECOND,
  millisToStringWithMaxPrecision,
  millisToStringWithSetPrecision,
  splitMillisIntoTimeUnits,
  timeUnitsToString,
  toOSDTime
} from '../../src/ts/util/time-util';

describe('Time utils', () => {
  const millis =
    MILLIS_IN_WEEK +
    MILLIS_IN_DAY +
    MILLIS_IN_MINUTE +
    // tslint:disable-next-line: no-magic-numbers
    MILLISECONDS_IN_SECOND * 30;

  test('Can split seconds', () => {
    const split = splitMillisIntoTimeUnits(millis);
    expect(split).toMatchSnapshot();
  });
  test('Can put time units into a high precision string', () => {
    const split = splitMillisIntoTimeUnits(millis);
    const asString = timeUnitsToString(split);
    expect(asString).toMatchSnapshot();
  });
  test('Can put time units into a fixed precision string', () => {
    const asString = millisToStringWithSetPrecision(millis, 2);
    expect(asString).toMatchSnapshot(asString);
  });
  test('Can put time units into a max precision string', () => {
    expect(millisToStringWithMaxPrecision(MILLIS_IN_MINUTE, 2));
  });

  test('Can convert to osd time', () => {
    const secondsInMinute = 60;
    expect(toOSDTime(secondsInMinute)).toMatchSnapshot();
  });
});
