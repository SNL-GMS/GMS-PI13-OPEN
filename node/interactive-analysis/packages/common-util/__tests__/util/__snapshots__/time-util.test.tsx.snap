// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time utils Can convert to osd time 1`] = `"1970-01-01T00:01:00.000Z"`;

exports[`Time utils Can put time units into a fixed precision string: 1 week 1 day 0 hour  1`] = `"1 week 1 day 0 hour "`;

exports[`Time utils Can put time units into a high precision string 1`] = `"1 week 1 day 1 minute 30 seconds"`;

exports[`Time utils Can split seconds 1`] = `
Object {
  "days": 1,
  "hours": 0,
  "milliseconds": 0,
  "minutes": 1,
  "seconds": 30,
  "weeks": 1,
}
`;
